torchao/prototype/mx_formats/mx_ops.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict
+
+import torch
+from torch.utils._pytree import tree_map
+
+from torchao.prototype.mx_formats.constants import DTYPE_FP4
+from torchao.prototype.mx_formats.mx_tensor import (  # noqa: E501
+    MXTensor,
+    tensor_size_hp_to_fp4x2,
+)
+
+aten = torch.ops.aten
+
+MX_OPS_TABLE: Dict[Any, Any] = {}
+
+
+def implements(aten_ops):
+    """Register aten ops to the mx op table"""
+
+    def decorator(func):
+        for op in aten_ops:
+            MX_OPS_TABLE[op] = func
+        return func
+
+    return decorator
+
+
+@implements([aten.detach.default])
```

Comment:
n00b: the spec primarly mentions that dot product and general dot product need to be implemented. All the below ops are sensible but this file seems sto do what's practical and not what the spec describes exactly

Reply:
we can add a docblock, these ops are not related to the MX spec and are related to what we need to support to enable training and inference if the MX spec is implemented with an `MXTensor` tensor subclass.